Sources/Paywalls/Components/Common/PaywallComponentPropertyTypes.swift

```diff
@@ -43,18 +43,80 @@ public extension PaywallComponent {
         public let heicLowRes: URL
     }
 
-    struct ColorInfo: Codable, Sendable, Hashable, Equatable {
+    struct ColorScheme: Codable, Sendable, Hashable, Equatable {
 
-        public init(light: ColorHex, dark: ColorHex? = nil) {
+        public init(light: ColorInfo, dark: ColorInfo? = nil) {
             self.light = light
             self.dark = dark
         }
 
-        public let light: ColorHex
-        public let dark: ColorHex?
+        public let light: ColorInfo
+        public let dark: ColorInfo?
 
     }
 
+    enum ColorInfo: Codable, Sendable, Hashable {
+
+        case hex(ColorHex)
+        case alias(String)
+
+        public func encode(to encoder: any Encoder) throws {
+            var container = encoder.container(keyedBy: CodingKeys.self)
+
+            switch self {
+            case .hex(let hex):
+                try container.encode(ColorInfoTypes.hex.rawValue, forKey: .type)
+                try container.encode(hex, forKey: .value)
+            case .alias(let alias):
+                try container.encode(ColorInfoTypes.alias.rawValue, forKey: .type)
+                try container.encode(alias, forKey: .value)
+            }
+        }
+
+        public init(from decoder: Decoder) throws {
```

Comment:
I might think about adding in some tests to validate the custom encoding/decoding of the ColorInfo enum. I know from recent experience in Pines that it's easy to make mistakes in these custom enum codable implementations and it's easy to miss until much later in testing

Reply:
Yes yes, for sure! Tests coming soon™️